asapdiscovery-genetics/asapdiscovery/genetics/seq_alignment.py

```diff
@@ -337,59 +443,164 @@ def do_MSA(
 
-    return select_file, p
+    alignment.sucess = True
+    return alignment
 
 
 # Defining colors for each protein residue
 def get_colors_protein(seqs):
     """Make colors for bases in sequence
 
-    Args:
-        seqs (list, str): List or string with protein sequence
+    Parameters
+    ----------
+    seq : List[str]
+        List with protein sequences
 
-    Returns:
-        list: List with colors
+    Returns
+    -------
+    list[str]
+        List with colors
     """
+    # List of all aminoacids in the list of seqs, in the same list
     text = [i for s in list(seqs) for i in s]
-    aa_colors = {
-        "A": "red",  # Alanine
-        "R": "blue",  # Arginine
-        "N": "green",  # Asparagine
-        "D": "yellow",  # Aspartic acid
-        "C": "orange",  # Cysteine
-        "Q": "purple",  # Glutamine
-        "E": "cyan",  # Glutamic acid
-        "G": "magenta",  # Glycine
-        "H": "pink",  # Histidine
-        "I": "brown",  # Isoleucine
-        "L": "gray",  # Leucine
-        "K": "lime",  # Lysine
-        "M": "teal",  # Methionine
-        "F": "navy",  # Phenylalanine
-        "P": "olive",  # Proline
-        "S": "maroon",  # Serine
-        "T": "silver",  # Threonine
-        "W": "gold",  # Tryptophan
-        "Y": "skyblue",  # Tyrosine
-        "V": "violet",  # Valine
-        "-": "white",
-    }
-    colors = [aa_colors[i] for i in text]
+    colors = []
+    for aa in text:
+        amino = AAcid(aa)
+        colors.append(amino.get_aminoacid_color())
     return colors
+
+
+# Defining colors for each protein residue
+def get_colors_by_aa_group(seq: str, max_missmatch=2):
+    """Make fill and text color for exact and group aminoacid matches
+
+    Parameters
+    ----------
+    seq : str
+        String with protein sequence
+    max_missmatch : int, optional
+       Maximum number of group missmatches after which match won't be highlighted, by default 2
+
+    Returns
+    -------
+    str, list[str]
+        Fill color, Font colors
+    """
+
+    from collections import Counter
+
+    seq_len = len(seq)
+    aa_groups = [AAcid(aa).get_aminoacid_group() for aa in seq]
+    aa_counts = Counter(aa_groups)
+    max_group, max_group_count = aa_counts.most_common(1)[0]
+    # Default font color
+    font_color = ["black"] * seq_len
+    # Check the case where all aa's are the same
+    if seq == seq_len * seq[0]:
+        color = "red"
+    # Check the case where all aa's belong to the same group (with some max missmatches)
+    elif max_group_count >= seq_len - max_missmatch:
+        if max_group is None:  # In case most "matches" are gaps
+            color = "white"
+        else:
+            color = "yellow"
+            # Make font red for missmatches
+            font_color = ["black" if item == max_group else "red" for item in aa_groups]
+    else:
+        color = "white"
+    return color, font_color
+
+
+class AAcid:
```

Comment:
This is nice, but simpler would probably to use `BioPython.seq1` or `BioPython.seq3` instead of MDAnalysis.